src/plugins/matrixops/vsplit_operation.cpp

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2018 Maxwell Reeser
+// Copyright (c) 2018 Hartmut Kaiser
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/execution_tree/primitives/base_primitive.hpp>
+#include <phylanx/execution_tree/primitives/slice.hpp>
+#include <phylanx/execution_tree/primitives/slice_node_data.hpp>
+#include <phylanx/execution_tree/primitives/slice_range.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/ir/ranges.hpp>
+#include <phylanx/plugins/matrixops/vsplit_operation.hpp>
+#include <phylanx/util/generate_error_message.hpp>
+
+#include <hpx/util/assert.hpp>
+
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace phylanx { namespace execution_tree { namespace primitives {
+
+    match_pattern_type const vsplit_operation::match_data = {
+        match_pattern_type{"vsplit", std::vector<std::string>{"vsplit(_1,_2)"},
+            &create_vsplit_operation, &create_primitive<vsplit_operation>, R"(
+            args
+            Args:
+
+                *args (m - matrix or column vector, N - number of blocks)
+
+            Returns:
+
+            An N by 1 matrix of vertical partitions of m)",
+            true}};
+
+    ///////////////////////////////////////////////////////////////////////////
+
+    vsplit_operation::vsplit_operation(primitive_arguments_type&& operands,
+        std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+      , dtype_(extract_dtype(name_))
+    {
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+
+    template <typename T>
+    primitive_argument_type vsplit_operation::vsplit2d_helper(
+        primitive_arguments_type&& args) const
+    {
+        std::array<std::size_t, 3> matrix_dims =
+            extract_numeric_value_dimensions(args[0], name_, codename_);
+
+        std::size_t num_rows = matrix_dims[0];
+        std::size_t num_cols = matrix_dims[1];
+
+        std::size_t block_dims =
+            extract_numeric_value_dimension(args[1], name_, codename_);
+
+        std::vector<std::pair<double, double>> ranges;
+
+        if (block_dims < 1)    // Scalar tiling
+        {
+            phylanx::ir::node_data<double> val =
+                extract_numeric_value(args[1], name_, codename_);
+
+            std::int64_t num_blocks = val.scalar();
+            //    extract_numeric_value(args[1], name_, codename_).scalar();
+            if (num_blocks <= 0)
```

Comment:
FWIW, you can use `extract_scalar_numeric_value(…)` to ensure the target is actually a scalar (will throw otherwise).